### ISSUE TITLE
S105: Exclude variable names containing `token` but also `suffix` or `prefix`

### ISSUE THREAD

**Issue by ThiefMaster** [astral-sh/ruff#8789]
https://github.com/astral-sh/ruff/blob/main/crates/ruff_linter/src/rules/flake8_bandit/helpers.rs

For example, it currently triggers on these 3 definitions:

```py
# The prefix for OAuth tokens
TOKEN_PREFIX_OAUTH = 'indo_'
# The prefix for personal tokens
TOKEN_PREFIX_PERSONAL = 'indp_'
# The prefix for service tokens (not handled by this module)
TOKEN_PREFIX_SERVICE = 'inds_'
```

However, clearly those are not hardcoded tokens and having suffixes/prefixes for tokens (especially since github's secret scanning becamea thing) is quite common, so maybe it'd be nice to resemble this in the regex that checks for token/password-like variable names instead of just noqa'ing it.

**Comment by charliermarsh** [comment]
For security-oriented checks, I would personally accept some false positives over risking false negatives. It seems reasonable to me to require users to explicitly `# noqa: S105` these, but I'd like to hear others' opinions.

**Comment by zanieb** [comment]
I agree I prefer false positives over negatives in security related rules. I wouldn't be _opposed_ to an exception for`PREFIX` and `SUFFIX` unless there's a case someone has where it'd cause a false negative.

**Comment by ThiefMaster** [comment]
A regex setting to whitelist variable names (and dict keys!) for "possible plaintext token/password" would be excellent :)

**Comment by IdoPort** [comment]
We are also experiencing some issues like that but from another angle.
We are  writing parsing code for several kind of files and we have functions that are looking or handling some behavior of specific tokens in the file.
In those functions we get FP alerts for each such constant so if we could suppress those by having some prefix to the `token` it would save us many `#noqa`.

**Comment by MichaReiser** [comment]
@IdoPort have you considered disabling the rule for those files? It seems very unlikely that they'll ever contain any real authentication token.

**Comment by IdoPort** [comment]
@MichaReiser I didn't know I can disable lints per file, that awesome! Thank!
